File: assignments/three/ElectricCar.py
from Car import Car
class ElectricCar (Car):
    def __init__(self, makeAndModel="none assigned", numberOfDoors=4, maximumNumberOfPassengers=5, batterySize=-1):
        super().__init__(makeAndModel,numberOfDoors,maximumNumberOfPassengers)
        self.batterySize = batterySize
    def setBatterySize(self,newBatterySize:int):
        self.batterySize = newBatterySize
    def getBatterySize(self)->int:
        return self.batterySize
    def __str__(self):
        s = super(ElectricCar,self).__str__()
        s += ", Battery Size: "+ str(self.getBatterySize())
        return s

if __name__=="__main__":
    c1 = ElectricCar("Tesla X 2023",2,4,10)
    print(c1)

File: assignments/four/ElectricCar.java

public class ElectricCar extends Car{
    private double batterySize;
    public ElectricCar(String makeAndModel,int maximumNumberOfPassangers, int numberOfDoors,double batterySize){
        super(makeAndModel,maximumNumberOfPassangers,numberOfDoors);
        this.batterySize = batterySize;
    }
    public double getBatterySize(){
        return batterySize;
    }
    public void setBatterySize(double batterySize){
        this.batterySize = batterySize;
    }
    public String toString(){
        return super.toString() + "\nBattery Size: " + batterySize;
    }
}
/*
ElectricCar: (11 points total)
    ElectricCar(String makeAndModel, int maximumNumberOfPassengers, int numberOfDoors, 
       double batterySize)  5 points
         toString()→ string 3 points
setBatterySize(double b) 2 point
            getBatterySize()→ double 1 poin
 */